### PR TITLE
Enable to pass filterRepoAddon function as optional params in add-in-repo-tests-to-host

### PR DESCRIPTION
**Pull request by choheekim**
It allows customize filtering for test trees.

In our use case, we'd want to preprocess our tests (using a Filter), e.g. adding attributes to testMetadata and so on, before test file paths are converted to AMD module paths.

### CODE REVIEW

**Comment by step2yeung** [comment]
Could we add a test to `ember-add-in-repo-tests/test/add-tests-to-host-test.js` for this?

**Comment by sangm** [comment]
Adding a test would document why the param would be useful

File: package.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
   "dependencies": {
     "broccoli-funnel": "^2.0.1",
     "broccoli-merge-trees": "^2.0.0",
+    "broccoli-persistent-filter": "^2.3.1",
```
Comment (benblank) [inline package.json]: This can go in `devDependencies`, as it's only used by tests.

**Comment by step2yeung** [comment]
@choheekim Thanks for adding the test and moving the dependency! 
@sangm mind giving this another look when you have the chance?

File: lib/add-in-repo-tests-to-host.js [hunk 2]
```diff
@@ -5,24 +5,29 @@ const mergeTrees = require('broccoli-merge-trees');
...
-const getInRepoAddonTestTrees = inRepoAddons => {
+const generateTreeFromPath = (path, processTests) =>
+  existsSync(path) ? new Funnel(processTests(path)) : null;
+
+const getInRepoAddonTestTrees = (inRepoAddons, processTests) => {
   if (inRepoAddons.length === 0) {
     return [null];
   }
 
-  return inRepoAddons.map(inRepoAddon => {
```
Comment (sangm) [inline lib/add-in-repo-tests-to-host.js:16]: Can you undo these changes?

File: lib/add-in-repo-tests-to-host.js [hunk 3]
```diff
@@ -5,24 +5,29 @@ const mergeTrees = require('broccoli-merge-trees');
...
     return existsSync(inRepoAddonTestPath)
-      ? new Funnel(inRepoAddonTestPath, { destDir: inRepoAddon.name })
+      ? new Funnel(processTests(inRepoAddonTestPath), {
+          destDir: inRepoAddon.name,
+        })
       : null;
   });
 };
 
-module.exports = (emberProject, predicate) => {
```
Comment (sangm) [inline lib/add-in-repo-tests-to-host.js:25]: nit: In some languages, underscore `_` is used to denote param will be unused. 

`processTests = _ => _` may be more clear for readability

File: lib/add-in-repo-tests-to-host.js [hunk 4]
```diff
@@ -32,16 +37,19 @@ module.exports = (emberProject, predicate) => {
   // if two addons have a common dependency
   const uniqueFlattenedAddons = flattenedAddons.filter(
     (flattenedAddon, index, allAddons) =>
-      allAddons.findIndex(addon => addon.name === flattenedAddon.name) === index
+      allAddons.findIndex((addon) => addon.name === flattenedAddon.name) ===
```
Comment (sangm) [inline lib/add-in-repo-tests-to-host.js:40]: Can you undo this change?

File: test/add-tests-to-host-test.js [hunk 5]
```diff
@@ -2,6 +2,7 @@
 
 const { expect } = require('chai');
 const { createTempDir, buildOutput } = require('broccoli-test-helper');
+const Filter = require('broccoli-persistent-filter');
```
Comment (sangm) [inline test/add-tests-to-host-test.js:5]: Is this dependency only used in tests? Seems like we can use a simpler function to unit test

Comment (choheekim) [reply test/add-tests-to-host-test.js:5]: it is. I've been trying to write the test without importing `broccoli-persistent-filter`. but we will still need external addon to read the file, e.g. glob and readFileSync.

Comment (choheekim) [reply test/add-tests-to-host-test.js]: Updated! @sangm do you mind giving a round of review?

File: lib/add-in-repo-tests-to-host.js [hunk 6]
```diff
@@ -1,14 +1,13 @@
...
 const { existsSync } = require('fs');
-const { WatchedDir } = require('broccoli-source');
 const mergeTrees = require('broccoli-merge-trees');
 const Funnel = require('broccoli-funnel');
 const flattenAddons = require('./flatten-addons');
 
-const generateTreeFromPath = path =>
-  existsSync(path) ? new WatchedDir(path) : null;
+const generateTreeFromPath = (path, processTests) =>
+  existsSync(path) ? new Funnel(processTests(path)) : null;
```
Comment (benblank) [inline lib/add-in-repo-tests-to-host.js:8]: Is this funnel necessary? It isn't doing anything, and looks like it could be replaced with just `existsSync(path) ? processTests(path) : null`. Also, why remove `WatchedDir`? Was what it was doing now unnecessary?

**Comment by step2yeung** [comment]
This looks good to me! Thanks for making all the changes @choheekim!

**Comment by sangm** [comment]
Looks good to me other than 1 nit.

**Comment by choheekim** [comment]
Resolved the last comment and squash all commits into one
